File: javascript/leetcode/1389. Create Target Array in the Given Order.js
/**
 * @param {number[]} nums
 * @param {number[]} index
 * @return {number[]}
 */
var createTargetArray = function (nums, index) {
  const result = [];
  let count = 0;

  for (let i of index) {
    result.splice(i, 0, nums[count++]);
  }

  return result;
};

File: c++/leetcode/1389. Create Target Array in the Given Order.cpp
class Solution
{
public:
  vector<int> createTargetArray(vector<int> &nums, vector<int> &index)
  {
    vector<int> result;
    int count = 0;

    for (auto i : index)
    {
      auto pos = result.begin();
      result.insert(pos + i, nums[count++]);
    }

    return result;
  }
};